packs/device42/actions/device_name_list.py

```diff
@@ -0,0 +1,70 @@
+import requests
+from st2actions.runners.pythonrunner import Action
+
+
+class DeviceNameList(Action):
+    def run(self, type=None, service_level=None, in_service=None, customer=None, tags=None,
+            blade_host_name=None, virtual_host_name=None, building_id=None, building=None,
+            room_id=None, room=None, rack_id=None, rack=None, serial_no=None,
+            serial_no_contains=None, asset_no=None, name=None, tags_and=None, uuid=None,
+            is_it_switch=None, is_it_virtual_host=None, is_it_blade_host=None, hardware=None,
+            hardware_ids=None, os=None, virtual_subtype=None, last_updated_lt=None,
+            last_updated_gt=None, first_added_lt=None, first_added_gt=None,
+            custom_fields_and=None, custom_fields_or=None):
+
+        d42_server = self.config.get('d42_server', None)
+        if not d42_server:
+            raise ValueError('"d42_server" config value is required')
+
+        d42_username = self.config.get('d42_username', None)
+        if not d42_username:
+            raise ValueError('"d42_username" config value is required')
+
+        d42_password = self.config.get('d42_password', None)
+        if not d42_password:
+            raise ValueError('"d42_password" config value is required')
+
+        verify = False
+        if self.config.get('verify_certificate', None) == 'true':
```

Comment:
YAML supports multiple types so the value specified by a user should be a boolean and not a string (as long as user specifies true / false for the attribute value).